### ISSUE TITLE
Various requirements for SAML

### ISSUE THREAD

**Issue by dcramer** [getsentry/sentry#5370]
Currently, SAML is blocked on a few things.

What we need is:

- [x] The ability to configure an SSO provider separately from identifying the user
- [x] Disabled "require SSO" until at least one owner has a linked identity.
  - this is needed to ensure that once an SSO provider is configured the user can still access things before identifying themselves
- [ ] Accessible method for logging into an organization when require SSO is not active (right now you can only do it if you mangle the URL)
  - Once this is done, we can also expose an option for "allow owners to bypass SSO"
- [x] csrf_exempt/protect on a per-view basis (per view in the pipeline), as some SAML views need csrf_excempt on POST
- [x] A way to logout the user's sessions when receiving the SAML logout action

**Comment by dcramer** [comment]
@mattrobenolt @mitsuhiko we need to figure out how to proceed with the above. The configuration flow changes are likely going to be a bit tricky. The rest is straight forward.

**Comment by pitbulk** [comment]
Also, in the oauth2 process, after setup a provider, mails are sent to users... in the saml2 flow, that is not required.

**Comment by dcramer** [comment]
@pitbulk wouldn't we still need to worry about merging existing accounts once saml is enabled? that's the primary reason for the emails

**Comment by pitbulk** [comment]
The "Link" can be done on the fly, when the user authenticates on the IdP, the IdP returns the SAMLResponse with user data, on the SAML settings option we can decide how link that data with a account that already exists (by username, by mail...)...this is how SAML integrations works.. we don't require a logged user to authenticate on the IdP in order to link accounts...

I think the process is different than with oauth2. oauth2 send an id, and you link that id with the sentry account... to be able to authenticate it in the future..

**Comment by mitsuhiko** [comment]
One of those items is implemented "A way to logout the user's sessions when receiving the SAML logout action". I don't know about the other one.

**Comment by EvanPurkhiser** [comment]
The last unchecked item can be broken out into its own issue should we ever want to prioritize that. Otherwise, these requirements are done.
